src/daft-io/src/s3_like.rs

```diff
@@ -633,6 +639,34 @@ async fn build_s3_conf(config: &S3Config) -> super::Result<s3::Config> {
     let force_path_style = config.endpoint_url.is_some() && !config.force_virtual_addressing;
     builder = builder.force_path_style(force_path_style);
 
+    let retry_unexpected_eof = {
+        #[derive(Debug)]
+        struct RetryUnexpectedEof;
+
+        impl ClassifyRetry for RetryUnexpectedEof {
+            fn classify_retry(&self, ctx: &InterceptorContext) -> RetryAction {
+                if let Some(Err(err)) = ctx.output_or_error()
+                    && format!("{err:?}").contains("UnexpectedEof")
```

Comment:
Oh interesting... what's the reasoning behind string matching instead of using the error's type?

Reply:
I know it's a bit of a hack but it's really hard to discern what the actual error type is based on the AWS library outputs. The actual `UnexpectedEof` error is nested in a series of `Box<Error>`s and I also couldn't find the crate that some of the error types were from. 

I could spend more time digging through it but I'm not certain I'd get anywhere anyway since I can't reproduce the error myself. Since `UnexpectedEof` is the actual name of the error type, it should not change unless we upgrade any dependencies, so I figured it would be fine to just match on the name.